pinot-query-runtime/src/test/java/org/apache/pinot/query/runtime/queries/ResourceBasedQueriesTest.java

```diff
@@ -375,9 +375,13 @@ private static Object[][] testResourceQueryTestCaseProviderWithMetadata()
               : replaceTableName(testCaseName, queryCase._sql);
 
           int segmentCount = 0;
-          for (String tableName : testCaseEntry.getValue()._tables.keySet()) {
-            segmentCount +=
-                _tableToSegmentMap.getOrDefault(testCaseName + "_" + tableName + "_OFFLINE", new HashSet<>()).size();
+          if (queryCase._expectedNumSegments != null) {
```

Comment:
what is this assert specifically doing? is it to test the segment level pruner?

Reply:
The thing is that this query has two joins so one table is scanned twice.

Reply:
hmm. I am not sure this is the right way to test. a simple final result check should already be sufficient (if not we should make the test case so that it is impossible to do it other than running 2 joins)

Reply:
done